assets/js/components/legacy-notifications/util.js

```diff
@@ -87,10 +88,15 @@ const removeDismissed = ( notifications ) => {
 		return notifications;
 	}
 
-	return notifications.filter( ( notification ) => {
-		const dismissed = getCache( `notification::dismissed::${ notification.id }` );
+	const promises = await notifications.map( ( notification ) => getItem( `notification::dismissed::${ notification.id }` ) );
+	const allNotifications = await Promise.all( promises );
+
+	const notDismissed = notifications.filter( ( _, index ) => {
```

Comment:
IMO it's more clear to return the result of the filtered notifications here than `notDismissed` .
```suggestion
	return notifications.filter( ( _, index ) => {
```